beacon-chain/blockchain/service.go

```diff
@@ -63,29 +60,58 @@ func NewChainService(ctx context.Context, cfg *Config) (*ChainService, error) {
 		beaconDB:           cfg.BeaconDB,
 		web3Service:        cfg.Web3Service,
 		incomingBlockChan:  make(chan *pb.BeaconBlock, cfg.IncomingBlockBuf),
-		processedBlockChan: make(chan *pb.BeaconBlock),
+		genesisTimeChan:    make(chan time.Time),
 		incomingBlockFeed:  new(event.Feed),
 		canonicalBlockFeed: new(event.Feed),
 		canonicalStateFeed: new(event.Feed),
-		unProcessedBlocks:  make(map[uint64]*pb.BeaconBlock),
-		unfinalizedBlocks:  make(map[[32]byte]*pb.BeaconState),
 		enablePOWChain:     cfg.EnablePOWChain,
 	}, nil
 }
 
 // Start a blockchain service's main event loop.
 func (c *ChainService) Start() {
-	log.Info("Starting service")
-	var err error
-	c.genesisTime, err = c.beaconDB.GenesisTime()
+	beaconState, err := c.beaconDB.State()
 	if err != nil {
-		log.Fatalf("Unable to retrieve genesis time - blockchain service could not start: %v", err)
-		return
+		log.Fatalf("Could not attempt fetch beacon state: %v", err)
+	}
+	// If the chain has already been initialized, simply start the block processing routine.
+	if beaconState != nil {
```

Comment:
So  `beaconstate == nil` will never happen because in startup the db will initialize a new state if none exists

Reply:
above im checking != nil

Reply:
Yeah thats what I mean, there is no point checking for that because `beaconstate != nil` will always  be true, since the db initializes a new state at startup https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/node/node.go#L175

Reply:
ah, k , sorry I didnt see that it had been removed